### DISCUSSION TITLE
Have an interactive mode

### DISCUSSION THREAD

**Discussion by SuspiciousActivity** [liquidctl/liquidctl#584] [Ideas]
I've come across this program because I'm using [this](https://github.com/jmarucha/FanControl.Liquidctl/). It's using liquidctl as an api to constantly check the status of supported devices every second, and change speeds accordingly.
The problem is the implementation: since there is no other way to get this information from CLI, it starts a new process every second, which is quite resource heavy.
I made a [fork](https://github.com/SuspiciousActivity/liquidctl/) of liquidctl specifically for this issue, and added an [interactive mode](https://github.com/SuspiciousActivity/liquidctl/commit/837ee5deaf1f6012aa6e97c58f6cfb6654b30278): you select one device (using --address/-d/--usb-port...) and you get one process that waits for input from stdin, like `status`, and then prints, in JSON and in a single line, the output of the status. Then it waits for more input.
My way seems really hacky though and I really had no idea what I was doing, but it works very well.
I made some other changes as well, so I didn't want to make the effort of creating a pull request without all that, just wanted to ask if you think this is a good idea and could be added to liquidctl?

**Comment by jonasmalacofilho** [comment]
A hacked-on interactive mode isn't a great fit for the main `liquidctl` CLI, but like an additional tool it makes sense.

We don't quite have an interactive mode, but in the `extra/` directory you can find [`liquiddump`](https://github.com/liquidctl/liquidctl/blob/main/extra/liquiddump), which was made for similar use cases (feeding status information from liquidctl to other programs).

Please let me know if it works well in practice, and feel free to send a PR with any improvements you find necessary (including a proper interactive mode, if that's really necessary).

**Reply by SuspiciousActivity** [reply]
We also need to set pump speeds through the same program, so that wouldn't work immediately. But I think I could just make another extra script which does what I described above, an make a PR for that. Didn't know liquiddump existed, thanks!
